src/engraving/rendering/dev/systemlayout.cpp

```diff
@@ -2576,7 +2565,12 @@ void SystemLayout::setInstrumentNames(System* system, LayoutContext& ctx, bool l
         return;
     }
     if (!ctx.conf().isShowInstrumentNames()
-        || (ctx.conf().styleB(Sid::hideInstrumentNameIfOneInstrument) && ctx.dom().visiblePartCount() <= 1)) {
+        || (ctx.conf().styleB(Sid::hideInstrumentNameIfOneInstrument) && ctx.dom().visiblePartCount() <= 1)
+        || (ctx.state().firstSystem()
+            && ctx.conf().styleV(Sid::firstSystemInstNameVisibility).value<InstrumentLabelVisibility>() == InstrumentLabelVisibility::HIDE)
```

Comment:
The only suggestion I have is to try to simplify this (for example, by adding temp bool variables). You can do this in your next PR